Bento: Can't import CSS file with webpack v5

### Description

In #35446, CSS files were added to the Bento components distributed via npm.

This worked great for my setup using webpack v4, where I was able to simply import the CSS file like this:

```js
import '@ampproject/amp-base-carousel/dist/styles.css';
```

The reason for this seems to be that webpack v5 supports the `exports` field in `package.json`, which v4 did not. And `styles.css` is not present there. So webpack complains.

To fix this, the following export would need to be added to `exports`:

```json
{
	  "./styles.css": "./dist/styles.css"
}
```

Then the following import works:

```js
import '@ampproject/amp-base-carousel/styles.css';
```

### Reproduction Steps

Try to import `styles.css` from `@ampproject/amp-base-carousel` in a webpack v5 environment.

The easiest way is through something like `@wordpress/scripts` v18, which is what I was using.

Then, import the CSS file in the source script and run `npx wp-scripts build <entry>` to try to build the script.

### Relevant Logs

```shell
Module not found: Error: Package path ./dist/styles.css is not exported from package /path/to/node_modules/@ampproject/amp-base-carousel (see exports field in /path/to/node_modules/@ampproject/amp-base-carousel/package.json)
```


### Browser(s) Affected

_No response_

### OS(s) Affected

_No response_

### Device(s) Affected

_No response_

### AMP Version Affected

2108280007000

--- a/build-system/npm-publish/write-package-files.js
+++ b/build-system/npm-publish/write-package-files.js
@@ -9,6 +9,7 @@ const {getSemver} = require('./utils');
 const {log} = require('../common/logging');
 const {stat, writeFile} = require('fs/promises');
 const {valid} = require('semver');
+const path = require('path');
 
 /**
  * Determines whether to skip
@@ -27,16 +28,14 @@ async function shouldSkip() {
 /**
  * Returns relative paths to all the extension's CSS file
  *
- * @return {Promise<boolean>}
+ * @return {Promise<string[]>}
  */
 async function getStylesheets() {
-  const files = await fastGlob(
-    `extensions/${extension}/${extensionVersion}/dist/**/*.css`
-  );
-
-  return files.map((path) =>
-    path.replace(`extensions/${extension}/${extensionVersion}/dist/`, '')
-  );
+  const extDir = `extensions/${extension}/${extensionVersion}/dist`
+    .split('/')
+    .join(path.sep);
+  const files = await fastGlob(path.join(extDir, '**', '*.css'));
+  return files.map((file) => path.relative(extDir, file));
 }
 
 /**

--- a/build-system/npm-publish/write-package-files.js
+++ b/build-system/npm-publish/write-package-files.js
@@ -4,9 +4,10 @@
  */
 
 const [extension, ampVersion, extensionVersion] = process.argv.slice(2);
+const fastGlob = require('fast-glob');
 const {getSemver} = require('./utils');
 const {log} = require('../common/logging');
-const {readdir, stat, writeFile} = require('fs/promises');
+const {stat, writeFile} = require('fs/promises');
 const {valid} = require('semver');
 
 /**
@@ -24,14 +25,18 @@ async function shouldSkip() {
 }
 
 /**
- * Determines whether a given directory path contains CSS files
+ * Determines whether the current extension contains CSS files
  *
- * @param {string} path Path to a directory.
  * @return {Promise<boolean>}
  */
-async function hasStylesheets(path) {
-  const files = await readdir(path);
-  return files.some((file) => file.endsWith('.css'));
+async function hasStylesheets() {
+  return (
+    (
+      await fastGlob(
+        `extensions/${extension}/${extensionVersion}/dist/**/*.css`
+      )
+    ).length > 0
+  );
 }
 
 /**
@@ -65,9 +70,7 @@ async function writePackageJson() {
     },
   };
 
-  if (
-    await hasStylesheets(`extensions/${extension}/${extensionVersion}/dist`)
-  ) {
+  if (await hasStylesheets()) {
     exports['./css/*'] = './dist/*';
   }
 

--- a/build-system/npm-publish/write-package-files.js
+++ b/build-system/npm-publish/write-package-files.js
@@ -25,17 +25,17 @@ async function shouldSkip() {
 }
 
 /**
- * Determines whether the current extension contains CSS files
+ * Returns relative paths to all the extension's CSS file
  *
  * @return {Promise<boolean>}
  */
-async function hasStylesheets() {
-  return (
-    (
-      await fastGlob(
-        `extensions/${extension}/${extensionVersion}/dist/**/*.css`
-      )
-    ).length > 0
+async function getStylesheets() {
+  const files = await fastGlob(
+    `extensions/${extension}/${extensionVersion}/dist/**/*.css`
+  );
+
+  return files.map((path) =>
+    path.replace(`extensions/${extension}/${extensionVersion}/dist/`, '')
   );
 }
 
@@ -70,8 +70,8 @@ async function writePackageJson() {
     },
   };
 
-  if (await hasStylesheets()) {
-    exports['./css/*'] = './dist/*';
+  for (const stylesheet of await getStylesheets()) {
+    exports[`./${stylesheet}`] = `./dist/${stylesheet}`;
   }
 
   const json = {

--- a/build-system/npm-publish/write-package-files.js
+++ b/build-system/npm-publish/write-package-files.js
@@ -5,11 +5,11 @@
 
 const [extension, ampVersion, extensionVersion] = process.argv.slice(2);
 const fastGlob = require('fast-glob');
+const path = require('path');
 const {getSemver} = require('./utils');
 const {log} = require('../common/logging');
 const {stat, writeFile} = require('fs/promises');
 const {valid} = require('semver');
-const path = require('path');
 
 /**
  * Determines whether to skip

--- a/build-system/npm-publish/write-package-files.js
+++ b/build-system/npm-publish/write-package-files.js
@@ -6,7 +6,7 @@
 const [extension, ampVersion, extensionVersion] = process.argv.slice(2);
 const {getSemver} = require('./utils');
 const {log} = require('../common/logging');
-const {stat, writeFile} = require('fs/promises');
+const {readdir, stat, writeFile} = require('fs/promises');
 const {valid} = require('semver');
 
 /**
@@ -21,6 +21,17 @@ async function shouldSkip() {
     log(`${extension} ${extensionVersion} : skipping, does not exist`);
     return true;
   }
+}
+
+/**
+ * Determines whether a given directory path contains CSS files
+ *
+ * @param {string} path Path to a directory.
+ * @return {Promise<boolean>}
+ */
+async function hasStylesheets(path) {
+  const files = await readdir(path);
+  return files.some((file) => file.endsWith('.css'));
 }
 
 /**
@@ -42,6 +53,24 @@ async function writePackageJson() {
     return;
   }
 
+  const exports = {
+    '.': './preact',
+    './preact': {
+      import: './dist/component-preact.module.js',
+      require: './dist/component-preact.js',
+    },
+    './react': {
+      import: './dist/component-react.module.js',
+      require: './dist/component-react.js',
+    },
+  };
+
+  if (
+    await hasStylesheets(`extensions/${extension}/${extensionVersion}/dist`)
+  ) {
+    exports['./css/*'] = './dist/*';
+  }
+
   const json = {
     name: `@ampproject/${extension}`,
     version,
@@ -50,18 +79,7 @@ async function writePackageJson() {
     license: 'Apache-2.0',
     main: './dist/component-preact.js',
     module: './dist/component-preact.module.js',
-    exports: {
-      '.': './preact',
-      './preact': {
-        import: './dist/component-preact.module.js',
-        require: './dist/component-preact.js',
-      },
-      './react': {
-        import: './dist/component-react.module.js',
-        require: './dist/component-react.js',
-      },
-      './styles.css': './dist/styles.css',
-    },
+    exports,
     files: ['dist/*', 'react.js'],
     repository: {
       type: 'git',

--- a/build-system/npm-publish/write-package-files.js
+++ b/build-system/npm-publish/write-package-files.js
@@ -60,6 +60,7 @@ async function writePackageJson() {
         import: './dist/component-react.module.js',
         require: './dist/component-react.js',
       },
+      './styles.css': './dist/styles.css',
     },
     files: ['dist/*', 'react.js'],
     repository: {

